README.md

```diff
@@ -1,5 +1,234 @@
-# iota-sdk
+# IOTA SDK Library
 
 [![Coverage Status](https://coveralls.io/repos/github/iotaledger/iota-sdk/badge.svg?branch=develop)](https://coveralls.io/github/iotaledger/iota-sdk?branch=develop)
 
-WIP
+The IOTA SDK is a Rust-based project that provides a convenient and efficient way to interact with nodes in the
+Shimmer and IOTA networks running
+the [Stardust protocol](https://wiki.iota.org/shimmer/develop/explanations/what-is-stardust). It consists of two main
+modules: `client` and `wallet`.
+
+The `client` module is stateless. It aims to provide more flexibility and access to low-level functions.
+
+The `wallet` module is stateful, with a standardized interface for developers to build applications involving value
+transactions. It uses high-level functions that simplify everyday operations. It can optionally interact
+with [IOTA Stronghold](https://github.com/iotaledger/stronghold.rs/) for seed handling, storage, and state backup.
+
+## Table of Contents
+
+- [Requirements](#requirements)
+    - [Dependencies](#dependencies)
+- [Getting Started](#getting-started)
+    - [Install the IOTA SDK](#install-the-iota-sdk)
+    - [Usage](#usage)
+        - [Client](#client)
+        - [Wallet](#wallet)
+        - [Examples](#examples)
+    - [API Reference](#api-reference)
+- [Contribute](#contribute)
+- [License](#license)
+
+## Features
+
+- **Client module**: The `client` module in the IOTA SDK offers low-level functions that allow you to have
+  fine-grained control over your interactions with Shimmer nodes. It provides access to the underlying API endpoints
+  and enables advanced operations such as custom message construction and direct communication with the network.
+
+- **Wallet module**: The `wallet` module in the IOTA SDK provides high-level functions for managing accounts, generating
+  addresses, creating transactions, and interacting with the Shimmer network. It offers a user-friendly interface for
+  developers to build applications on the Shimmer network.
```

Comment:
This is somewhat duplicated with the first section, maybe merge them and remove one?